### PR TITLE
gui: fix pagination errors.

### LINKED ISSUE #292: Remove hardcoded localhost
`index.html` connection instructions have a hard-coded localhost. This should be dynamic to display the actual current hostname.

https://github.com/decred/dcrpool/blob/49f5ce59892f6dfaab440d79b3063e21a99e82a4/gui/assets/templates/index.html#L88

### PR DESCRIPTION
**Pull request by dnldd**
This fixes pagination related errors when the data source returns nil. The supported miner list has also been updated.

Closes #292

### CODE REVIEW

File: gui/index.go [hunk 1]
```diff
@@ -38,6 +38,13 @@ func (ui *GUI) renderIndex(w http.ResponseWriter, r *http.Request, modalError st
 	// time, but the GUI doesn't use them yet.
 	lastPaymentHeight, _, _ := ui.cache.getLastPaymentInfo()
 
+	var address string
+	if ui.cfg.UseLEHTTPS {
```
Comment (jholdstock) [inline gui/index.go:42]: Should use `if ui.cfg.Domain != nil` here.

If somebody is using another webserver like nginx to provide TLS, `ui.cfg.UseLeHTTPS` will be `false` and `127.0.0.1` will be incorrectly shown on the website.

**Review by jholdstock — CHANGES_REQUESTED** [review]

File: gui/index.go (hunk 1 above)
Comment (dnldd) [reply gui/index.go:42]: Good catch, fixing.

Comment (jholdstock) [reply gui/index.go]: I dont get why you've added the extra logic for UseLEHTTPS? Surely just `if ui.cfg.Domain != ""` is enough?

Comment (dnldd) [reply gui/index.go]: It's for the two cases where letsencrypt is being used and when its not, In the case where its not but TLS is still being provided then the domain would be set. In both cases the domain shown should be the domain set.

Comment (dnldd) [reply gui/index.go]: you're right, no need checking the letsencrypt flag to set the domain.

**Review by jholdstock — APPROVED** [review]

**Pull request merged**